envoy-control-core/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/snapshot/listeners/filters/RBACFilterFactory.kt

```diff
@@ -124,31 +141,44 @@ class RBACFilterFactory(
         return methodPermissions
     }
 
-    private fun resolveClients(incomingEndpoint: IncomingEndpoint, roles: List<Role>): List<String> {
-        val clients = incomingEndpoint.clients.flatMap { clientOrRole ->
+    private fun resolveClientComposites(incomingEndpoint: IncomingEndpoint, roles: List<Role>): List<ClientComposite> {
+        val clientComposites = incomingEndpoint.clients.flatMap { clientOrRole ->
```

Comment:
Is it possible to keep roles as `Map`? If so we won't have to do nested loop. It isn't necessary now.

Reply:
I think it could, WDYT @MarcinFalkowski ?

Reply:
The size of `roles` collection will be typically so small that I think creating and accessing the map will be less efficient than searching a list. But the difference will be not noticeable, so I have nothing against it if you prefer it.